airbyte-commons-worker/src/test/java/io/airbyte/workers/general/LimitedAirbyteSource.java

```diff
@@ -0,0 +1,46 @@
+package io.airbyte.workers.general;
+
+import io.airbyte.config.WorkerSourceConfig;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.workers.internal.AirbyteSource;
+import io.airbyte.workers.test_utils.AirbyteMessageUtils;
+import java.nio.file.Path;
+import java.util.Optional;
+
+public class LimitedAirbyteSource implements AirbyteSource {
+  private static final int TOTAL_RECORDS = 10_000_000;
+
+  private int currentRecords = 0;
+
+  @Override
+  public void start(WorkerSourceConfig sourceConfig, Path jobRoot) throws Exception {
+
+  }
+
+  @Override
+  public boolean isFinished() {
+    return currentRecords == TOTAL_RECORDS;
+  }
+
+  @Override
+  public int getExitValue() {
+    return 0;
+  }
+
+  @Override
+  public Optional<AirbyteMessage> attemptRead() {
+    currentRecords++;
+    return Optional.of(AirbyteMessageUtils.createRecordMessage("test stream", "data",
+        "This is a fairly long sentence to provide some bytes here. More bytes is better as it helps us measure performance."));
+  }
```

Comment:
You mentioned in slack that this might be optimized by the JVM... maybe append the timestamp to the string to test that?

But, either way, that's just in the READ process... so this being fast shouldn't make the orchestrator any faster...

Reply:
Good point!

Reply:
Append and not much change.